### PR TITLE
Bump Enzyme to v0.13

### PR DESCRIPTION
**Pull request by mhauru**
Closes #328 Closes #327 Closes #340

### CODE REVIEW

**Comment by penelopeysm** [comment]
@mhauru I took the liberty to force push at 2:30 am to undo my commits which I meant to push elsewhere, sorry 😬

File: test/ad/flows.jl [hunk 1]
```diff
@@ -1,26 +1,30 @@
 @testset "PlanarLayer" begin
+    # TODO(mhauru) Remove the EnzymeReverseCrash marks once this has been fixed:
+    # https://github.com/EnzymeAD/Enzyme.jl/issues/2029
```
Comment (wsmoses) [inline test/ad/flows.jl:3]: these crashes and related should all be resolved

Comment (mhauru) [reply test/ad/flows.jl]: Marked as unbroken. Likewise marked a few others.

**Comment by devmotion** [comment]
Fixed by #350 which was based on and included this PR.

**Pull request closed without merging**